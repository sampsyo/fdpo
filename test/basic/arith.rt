in left: 32;
in right: 32;
out a: 32;
out s: 32;
out m: 32;
out d: 32;
out r: 32;
a = add[32](left, right);
s = sub[32](left, right);
m = mul[32](left, right);
d = div[32](left, right);
r = mod[32](left, right);
