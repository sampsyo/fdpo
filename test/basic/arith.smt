(let ((.def_0 (bvurem left right))) (let ((.def_1 (= r .def_0))) (let ((.def_2 (bvudiv left right))) (let ((.def_3 (= d .def_2))) (let ((.def_4 (bvmul left right))) (let ((.def_5 (= m .def_4))) (let ((.def_6 (bvsub left right))) (let ((.def_7 (= s .def_6))) (let ((.def_8 (bvadd left right))) (let ((.def_9 (= a .def_8))) (let ((.def_10 (and .def_9 .def_7 .def_5 .def_3 .def_1))) .def_10)))))))))))
