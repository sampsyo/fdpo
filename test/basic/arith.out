a = 47
s = 37
m = 210
d = 8
r = 2
